### PR TITLE
Add keyboard and keymap for The Nines macropad

### PR DESCRIPTION
**Pull request by luis-godinez**
Ungodly Design | The Nines Macropad.

## Description

Add keyboard and mapping.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by drashna — COMMENTED** [review]
Could you add a license header to all of the `c` and `h` files (including the keymaps? 

Aside from that, it looks good to me.

File: keyboards/ungodly/nines/config.h [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x5544 // "UD" = Ungodly Design
+#define PRODUCT_ID      0x544E // "TN" = The Nines
+#define DEVICE_VER      0x9999
+#define MANUFACTURER    Ungodly Design
+#define PRODUCT         The Nines by Ungodly Design
+#define DESCRIPTION     A 9 key (3x3) macropad with rotary encoder support.
```
Comment (fauxpark) [inline keyboards/ungodly/nines/config.h:27]:
```suggestion
```

File: keyboards/ungodly/nines/config.h [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x5544 // "UD" = Ungodly Design
+#define PRODUCT_ID      0x544E // "TN" = The Nines
+#define DEVICE_VER      0x9999
+#define MANUFACTURER    Ungodly Design
+#define PRODUCT         The Nines by Ungodly Design
```
Comment (fauxpark) [inline keyboards/ungodly/nines/config.h:26]:
```suggestion
#define PRODUCT         The Nines
```

File: keyboards/ungodly/nines/rules.mk [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+MCU = atmega32u4
+
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
```
Comment (fauxpark) [inline keyboards/ungodly/nines/rules.mk:11]:
```suggestion
```

File: keyboards/ungodly/nines/rules.mk [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes           # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality on B7 by default
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
+ENCODER_ENABLE = yes        # Rotary encoder support
```
Comment (fauxpark) [inline keyboards/ungodly/nines/rules.mk:33]:
```suggestion
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
ENCODER_ENABLE = yes        # Rotary encoder support
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by luis-godinez — COMMENTED** [review]
thanks. done.

File: keyboards/ungodly/nines/readme.md [hunk 5]
```diff
@@ -0,0 +1,16 @@
+# Nines
+
+![Nines](http://i.imgur.com/q2M3uEU.jpg)
```
Comment (fauxpark) [inline keyboards/ungodly/nines/readme.md:3]:
```suggestion
![Nines](https://i.imgur.com/q2M3uEUl.jpg)
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/ungodly/nines/readme.md (hunk 5 above)
Comment (luis-godinez) [reply keyboards/ungodly/nines/readme.md]: Updated url with correct link.

**Review by drashna — APPROVED** [review]

File: keyboards/ungodly/nines/readme.md [hunk 6]
```diff
@@ -0,0 +1,16 @@
+# Nines
+
+![Nines](https://www.instagram.com/p/CFFPIJBHDSp)
```
Comment (fauxpark) [inline keyboards/ungodly/nines/readme.md:3]: This doesn't render anymore. An imgur link is preferred.

File: keyboards/ungodly/nines/config.h [hunk 7]
```diff
@@ -0,0 +1,217 @@
...
+   - MIDI notes can be added to the keymap
+   - Octave shift and transpose
+   - Virtual sustain, portamento, and modulation wheel
+   - etc.
+*/
+//#define MIDI_ADVANCED
+
+/* override number of MIDI tone keycodes (each octave adds 12 keycodes and allocates 12 bytes) */
+//#define MIDI_TONE_KEYCODE_OCTAVES 1
+
```
Comment (fauxpark) [inline keyboards/ungodly/nines/config.h:214]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/ungodly/nines/readme.md [hunk 8]
```diff
@@ -0,0 +1,16 @@
+# Nines
+
+![Nines](https://imgur.com/a/EcypUBG)
```
Comment (noroadsleft) [inline keyboards/ungodly/nines/readme.md:3]: This needs to be a direct image link.

```suggestion
![Nines](https://i.imgur.com/zNnmPV9l.jpg)
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]

**Comment by noroadsleft** [comment]
@luis-godinez What was the reason for closing this pull request?

**Comment by luis-godinez** [comment]
I'll clean up the CL and request a pull in the future.

**Comment by noroadsleft** [comment]
You don't actually have to open a whole new pull request. When you open a pull request, GitHub tracks which branch contains your changes. All you have to do is make any requested changes and commit them to the same branch from which you opened the PR, then push the branch to GitHub.

**Pull request closed without merging**